Core/GDCore/IDE/Events/ExpressionsParameterMover.cpp

```diff
@@ -79,37 +79,43 @@ class GD_CORE_API ExpressionParameterMover
   void OnVisitObjectFunctionNameNode(ObjectFunctionNameNode& node) override {}
   void OnVisitFunctionCallNode(FunctionCallNode& node) override {
     auto moveParameter =
-        [this](std::vector<std::unique_ptr<gd::ExpressionNode>>& parameters) {
-          if (oldIndex >= parameters.size() || newIndex >= parameters.size())
+        [this](std::vector<std::unique_ptr<gd::ExpressionNode>>& parameters, int firstWrittenParameterIndex) {
+          size_t newExpressionIndex = newIndex - firstWrittenParameterIndex;
+          size_t oldExpressionIndex = oldIndex - firstWrittenParameterIndex;
+
+          if (oldExpressionIndex >= parameters.size() || newExpressionIndex >= parameters.size())
             return;
 
-          auto movedParameterNode = std::move(parameters[oldIndex]);
-          parameters.erase(parameters.begin() + oldIndex);
-          parameters.insert(parameters.begin() + newIndex,
+          auto movedParameterNode = std::move(parameters[oldExpressionIndex]);
+          parameters.erase(parameters.begin() + oldExpressionIndex);
+          parameters.insert(parameters.begin() + newExpressionIndex,
                             std::move(movedParameterNode));
         };
 
     if (node.functionName == functionName) {
       if (behaviorType.empty() && !objectType.empty() &&
           !node.objectName.empty()) {
         // Move parameter of an object function
+        // This refactor only applies on events functions
+        // and events object functions doesn't exist yet.
+        // This is a dead code.
```

Comment:
I'm no expert in this part of the code, but if this is dead code, should we remove the lines below and adapt the comment saying this can't happen?

Reply:
A word is missing, it should be "This refactor only applies on events [object] functions..."